### ISSUE TITLE
SC2032 and SC2033 false positives when conditionally defining function

### ISSUE THREAD

**Issue by tdmalone** [koalaman/shellcheck#1200]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2032/SC2033
- My shellcheck version (`shellcheck --version` or "online"): 0.4.7
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/usr/bin/env bash

if [ "$( sudo -v )" ]; then
  sudo mv terraform /usr/bin/terraform
else
  terraform() { "${TRAVIS_BUILD_DIR}/terraform" "$@"; }
  export -f terraform
fi

```

#### Here's what shellcheck currently says:

```
Line 4:
  sudo mv terraform /usr/bin/terraform
          ^-- SC2033: Shell functions can't be passed to external commands.
 
Line 6:
  terraform() { "${TRAVIS_BUILD_DIR}/terraform" "$@"; }
  ^-- SC2032: Use own script or sh -c '..' to run this from sudo.
```

#### Here's what I wanted or expected to see:

```
No issues detected!
```

**Comment by tdmalone** [comment]
I'm open to the fact that I might be doing something wrong, but I think this is a false positive. I'm conditionally creating the function, so it shouldn't even exist at the point I'm trying to run sudo instead.

I also realise this might be very hard to detect, and I will use a directive for now.

**Comment by Dmole** [comment]
I think if you or some calling script wrap your whole code in brackets it will create the function and export it then check the condition so I think it's a valid warning. You should probably just re-name your function.
